exasol_script_languages_container_tool/lib/models/language_activation.py

```diff
@@ -0,0 +1,68 @@
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import PurePosixPath
+from typing import List, Optional, Union
+from urllib.parse import ParseResult, urlencode, urlunparse
+
+
+class SLCLanguage(Enum):
+    Java = "java"
+    Python3 = "python"
+    R = "r"
+
+
+@dataclass
+class SLCParameter:
+    key: str
+    value: List[str]
+
+
+@dataclass
+class LanguageDefinitionURL:
+    protocol: str
+    chroot_bucketfs_name: str
+    chroot_bucket_name: str
+    udf_client_bucketfs_name: str
+    udf_client_bucket_name: str
+    parameters: List[SLCParameter]
+    language: SLCLanguage
+    chroot_path_in_bucket: Optional[PurePosixPath] = None
+    udf_client_executable: Optional[PurePosixPath] = None
+
+    def __str__(self) -> str:
+        query_params = {p.key: v for p in self.parameters for v in p.value}
+        query_params["lang"] = self.language.value.lower()
+        query_string = urlencode(query_params)
+        url = urlunparse(
+            ParseResult(
+                scheme=self.protocol,
+                netloc="",
+                path=f"///{self.chroot_bucketfs_name}/{self.chroot_bucket_name}/{self.chroot_path_in_bucket or ''}",
+                params="",
+                query=query_string,
+                fragment=f"buckets/{self.udf_client_bucketfs_name}/{self.udf_client_bucket_name}/"
```

Comment:
self.udf_client_bucketfs_name are optional
self.udf_client_bucket_name are optional

then there is also no buckets

Reply:
Done, implicitely by adding new dataclasses `UdfClientRelativePath` and `UdfClientAbsolutePath`